### PR TITLE
Initial Spaces feature flag

### PR DESCRIPTION
**Pull request by t3chguy**
Based on https://github.com/matrix-org/matrix-react-sdk/pull/5667

Requires https://github.com/matrix-org/matrix-js-sdk/pull/1610

### CODE REVIEW

File: src/settings/Settings.ts [hunk 1]
```diff
@@ -120,6 +120,13 @@ export interface ISetting {
 }
 
 export const SETTINGS: {[setting: string]: ISetting} = {
+    "feature_spaces": {
+        isFeature: true,
+        displayName: _td("Spaces prototype. Incompatible with Communities, Communities v2 and Custom Tags"),
```
Comment (jryans) [inline src/settings/Settings.ts:125]: If it's going to have full stops, let's use them for all phrases.

```suggestion
        displayName: _td("Spaces prototype. Incompatible with Communities, Communities v2 and Custom Tags."),
```

Comment (jryans) [reply src/settings/Settings.ts:125]: Should this say it requires a compatible homeserver, like the communities v2 flag?

File: src/settings/Settings.ts [hunk 2]
```diff
@@ -134,6 +141,7 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         ),
         supportedLevels: LEVELS_FEATURE,
         default: false,
+        controller: new IncompatibleController("feature_spaces"),
```
Comment (jryans) [inline src/settings/Settings.ts:144]: Is this module missing? I don't see it locally...

**Review by jryans — CHANGES_REQUESTED** [review]
Please also document the new flag in `element-web` as well.

File: src/settings/Settings.ts (hunk 1 above)
Comment (t3chguy) [reply src/settings/Settings.ts:125]: I guess so, partially. "Some parts of this feature require a compatible homeserver" - I'm all for making the description longer and making this Labs flag stand out even more :P

File: src/settings/Settings.ts (hunk 2 above)
Comment (t3chguy) [reply src/settings/Settings.ts:144]: As per the PR body

> Based on https://github.com/matrix-org/matrix-react-sdk/pull/5668

Comment (jryans) [reply src/settings/Settings.ts:144]: Ah huh, sorry, I guess I didn't notice / understand until now... 😅

**Review by jryans — APPROVED** [review]

**Pull request merged**